src/Features/Core/Portable/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -29,8 +29,6 @@ internal class MetadataAsSourceFileService : IMetadataAsSourceFileService
         /// </summary>
         private readonly SemaphoreSlim _gate = new(initialCount: 1);
 
-        private readonly Dictionary<string, IMetadataAsSourceFileProvider> _tempFileToProviderMap = new(StringComparer.OrdinalIgnoreCase);
```

Comment:
Slightly longer explanation for removing this is in the first commit, but TL;DR is:
* Unnecessary because only two providers
* Only allows one file per symbol which doesn't hold for PDB source